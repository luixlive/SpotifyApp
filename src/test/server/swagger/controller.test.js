import swaggerController from './../../../server/swagger/controller';

describe('Server Swagger - Controller', () => {
  let redirectTo;
  let header;
  let req;
  let res;
  let responseValue;
  beforeEach(() => {
    header = { header: undefined, value: undefined };
    redirectTo = undefined;
    responseValue = undefined;
    res = {
      redirect: (path) => { redirectTo = path; },
      send: (value) => { responseValue = value; },
      setHeader: (headerName, value) => {
        header = { header: headerName, value };
      },
    };
  });

  describe('Swagger Default', () => {
    it('redirects to /swagger/api-docs', () => {
      swaggerController.swaggerDefault(req, res);
      expect(redirectTo).toEqual('/swagger/api-docs');
    });
  });

  describe('Swagger JS Doc', () => {
    it('sets headers and sends Swagger JSON', () => {
      swaggerController.swaggerJSDoc(req, res);
      expect(header).toEqual({
        header: 'Content-Type',
        value: 'application/json',
      });
      expect(responseValue).toMatchSnapshot();
    });
  });
});
