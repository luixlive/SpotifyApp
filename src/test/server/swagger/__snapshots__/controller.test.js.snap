// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Server Swagger - Controller Swagger JS Doc sets headers and sends Swagger JSON 1`] = `
Object {
  "definitions": Object {
    "Artists": Object {
      "items": Object {
        "allOf": Array [
          Object {
            "$ref": "#/definitions/UserBase",
          },
          Object {
            "properties": Object {
              "genres": Object {
                "items": Object {
                  "type": "string",
                },
                "type": "array",
              },
              "name": Object {
                "type": "string",
              },
              "popularity": Object {
                "format": "int32",
                "type": "number",
              },
            },
            "type": "object",
          },
        ],
        "required": Array [
          "genres",
          "name",
          "popularity",
        ],
      },
      "type": "array",
    },
    "Error": Object {
      "properties": Object {
        "error": Object {
          "type": "string",
        },
      },
      "required": Array [
        "error",
      ],
      "type": "object",
    },
    "User": Object {
      "properties": Object {
        "accessToken": Object {
          "type": "string",
        },
        "profile": Object {
          "allOf": Array [
            Object {
              "$ref": "#/definitions/UserBase",
            },
            Object {
              "properties": Object {
                "displayName": Object {
                  "type": "string",
                },
              },
              "type": "object",
            },
          ],
        },
        "refreshToken": Object {
          "type": "string",
        },
      },
      "required": Array [
        "accessToken",
        "profile",
        "refreshToken",
      ],
      "type": "object",
    },
    "UserBase": Object {
      "properties": Object {
        "externalUrls": Object {
          "properties": Object {
            "spotify": Object {
              "format": "uri",
              "type": "string",
            },
          },
          "type": "object",
        },
        "followers": Object {
          "properties": Object {
            "href": Object {
              "format": "uri",
              "type": "string",
            },
            "total": Object {
              "format": "int64",
              "type": "number",
            },
          },
          "type": "object",
        },
        "href": Object {
          "format": "uri",
          "type": "string",
        },
        "id": Object {
          "type": "string",
        },
        "images": Object {
          "items": Object {
            "properties": Object {
              "height": Object {
                "format": "int32",
                "type": "number",
              },
              "url": Object {
                "format": "uri",
                "type": "string",
              },
              "width": Object {
                "format": "int32",
                "type": "number",
              },
            },
            "type": "object",
          },
          "type": "array",
        },
        "type": Object {
          "type": "string",
        },
        "uri": Object {
          "format": "uri",
          "type": "string",
        },
      },
      "required": Array [
        "externalUrls",
        "followers",
        "href",
        "id",
        "images",
        "type",
        "uri",
      ],
      "type": "object",
    },
  },
  "info": Object {
    "description": "Simple webapp to see your user's Spotify statistics.",
    "title": "Spotify Stats",
    "version": "1.0.0",
  },
  "parameters": Object {},
  "paths": Object {
    "/api/authentication/logout": Object {
      "get": Object {
        "description": "Logout from Spotify",
        "parameters": Array [
          Object {
            "default": "Set it in the browser, leave this field as it is",
            "description": "Session cookie signed",
            "in": "cookie",
            "name": "session",
            "schema": Object {
              "type": "string",
            },
          },
          Object {
            "default": "Set it in the browser, leave this field as it is",
            "description": "Signature key for the Session cookie",
            "in": "cookie",
            "name": "session.sig",
            "schema": Object {
              "type": "string",
            },
          },
        ],
        "responses": Object {
          "204": Object {
            "description": "User logged out",
          },
          "401": Object {
            "description": "No user found",
          },
        },
        "tags": Array [
          "Authentication",
        ],
      },
    },
    "/api/authentication/spotify": Object {
      "get": Object {
        "description": "Get redirected to Spotify's login page, after login event /api/authentication/spotify/callback will be called",
        "tags": Array [
          "Authentication",
        ],
      },
    },
    "/api/authentication/user": Object {
      "get": Object {
        "description": "Retrieve current user (if there is one)",
        "parameters": Array [
          Object {
            "default": "Set it in the browser, leave this field as it is",
            "description": "Session cookie signed",
            "in": "cookie",
            "name": "session",
            "schema": Object {
              "type": "string",
            },
          },
          Object {
            "default": "Set it in the browser, leave this field as it is",
            "description": "Signature key for the Session cookie",
            "in": "cookie",
            "name": "session.sig",
            "schema": Object {
              "type": "string",
            },
          },
        ],
        "produces": Array [
          "application/json",
        ],
        "responses": Object {
          "200": Object {
            "description": "User found and returned",
            "schema": Object {
              "$ref": "#/definitions/User",
            },
          },
          "401": Object {
            "description": "No user found",
          },
        },
        "tags": Array [
          "Authentication",
        ],
      },
    },
    "/api/stats/topArtists": Object {
      "get": Object {
        "description": "Retrieve from Spotify user's top artists",
        "parameters": Array [
          Object {
            "default": "Set it in the browser, leave this field as it is",
            "description": "Session cookie signed",
            "in": "cookie",
            "name": "session",
            "schema": Object {
              "type": "string",
            },
          },
          Object {
            "default": "Set it in the browser, leave this field as it is",
            "description": "Signature key for the Session cookie",
            "in": "cookie",
            "name": "session.sig",
            "schema": Object {
              "type": "string",
            },
          },
          Object {
            "default": 5,
            "description": "Number of artists (minimum 1, maximum 50)",
            "format": "int32",
            "in": "query",
            "maximum": 50,
            "minimum": 1,
            "name": "limit",
            "type": "number",
          },
          Object {
            "default": 0,
            "description": "Offset value (minimum 0)",
            "format": "int32",
            "in": "query",
            "minimum": 0,
            "name": "offset",
            "type": "number",
          },
          Object {
            "default": "medium_term",
            "description": "Time range to take into account artists",
            "enum": Array [
              "long_term",
              "medium_term",
              "short_term",
            ],
            "in": "query",
            "name": "timeRange",
            "type": "string",
          },
        ],
        "responses": Object {
          "200": Object {
            "description": "List of user's top artists",
            "schema": Object {
              "$ref": "#/definitions/Artists",
            },
          },
          "400": Object {
            "description": "Bad request",
            "schema": Object {
              "$ref": "#/definitions/Error",
            },
          },
          "401": Object {
            "description": "No user found",
          },
          "502": Object {
            "description": "Error retrieving artists from Spotify",
            "schema": Object {
              "$ref": "#/definitions/Error",
            },
          },
        },
        "tags": Array [
          "Stats",
        ],
      },
    },
  },
  "responses": Object {},
  "securityDefinitions": Object {},
  "swagger": "2.0",
  "tags": Array [
    Object {
      "name": "Authentication",
    },
    Object {
      "name": "Stats",
    },
  ],
}
`;
